In [30]:
import numpy as np 
import pandas as pd

In [31]:
#Read the data 
body_df = pd.read_csv('../Data/SolarSystemData.csv')
print(body_df.to_string())

   Id     Name                Type  Order from Sun  Diameter (Earth=1)  Diameter (km)  Mass (Earth=1)  Mean Distance from Sun (AU)  Orbital Period (years)  Orbital Eccentricity  Mean Orbital Velocity (km/sec)  Rotation Period (days)  Inclination of Axis (degrees)  Mean Temperature at Surface (C)  Gravity at Equator (Earth=1)  Escape Velocity (km/sec)  Mean Density (water=1) Atmospheric Composition  Number of Moons  Rings  Parent               Image                                            URI
0   1      Sun                Star               0             109.000        1392000      333000.000                         0.00                    0.00                0.0000                            0.00                   25.05                           7.25                             5600                         28.00                    617.70                    1.40                   H2+He                0  False     NaN      Images\Sun.jpg               http://en.wikipedia.org/wiki/Sun


In [75]:
# 1. Calculate the relative size and distances of the planets

# Define the size of the Sun in km
sun_radius_km = 696340

ds_dict = {}

# Loop over each row in the dataframe and calculate the size and distance of each planet relative to the Sun
for index, row in body_df.iterrows():
    # Calculate the size of the planet relative to the Sun
    planet_size = row['Diameter (km)'] / (sun_radius_km * 2)
    
    # Calculate the distance of the planet relative to the Sun
    planet_distance = row['Mean Distance from Sun (AU)'] * 149.6e6 / sun_radius_km

    # Add the results to the dictionary
    ds_dict[row['Name']] = [planet_size, planet_distance]
    
    # Print the results
    print(f"{row['Name']}: Radius - {planet_size:.5f} Solar radii, Distance - {planet_distance:.3f} Solar radii")

ds_dict

Sun: Radius - 0.99951 Solar radii, Distance - 0.000 Solar radii
Mercury: Radius - 0.00350 Solar radii, Distance - 83.787 Solar radii
Venus: Radius - 0.00869 Solar radii, Distance - 154.683 Solar radii
Earth: Radius - 0.00916 Solar radii, Distance - 214.838 Solar radii
Mars: Radius - 0.00487 Solar radii, Distance - 326.553 Solar radii
Jupiter: Radius - 0.10254 Solar radii, Distance - 1117.155 Solar radii
Saturn: Radius - 0.08616 Solar radii, Distance - 2049.551 Solar radii
Uranus: Radius - 0.03670 Solar radii, Distance - 4120.585 Solar radii
Neptune: Radius - 0.03556 Solar radii, Distance - 6458.018 Solar radii
Pluto: Radius - 0.00165 Solar radii, Distance - 8473.194 Solar radii


{'Sun': [0.9995117327742195, 0.0],
 'Mercury': [0.0035025993049372433, 83.78665594393543],
 'Venus': [0.008691156618893069, 154.6830571272654],
 'Earth': [0.00915931872361203, 214.83757934342418],
 'Mars': [0.004873337737312232, 326.55312060200475],
 'Jupiter': [0.102536117413907, 1117.1554125858058],
 'Saturn': [0.08616480454950168, 2049.5505069362666],
 'Uranus': [0.03670477065801189, 4120.584771806876],
 'Neptune': [0.035563086997730996, 6458.017635063331],
 'Pluto': [0.0016514920871987821, 8473.19412930465]}

### Which planets have anomalous eccentricities? Which have anomalous inclinations? Which have anomalous rotation periods?

- Anormal Orbital Eccentricities 
  - Both Mercury and Pluto have higher than average orbital eccentricities, meaning that they have more eliptical orbits. 
- Anormal Inclination of the Axis (I don't see an orbital inclination field in the csv)
  - There doesn't seem to be a real pattern with the inclination of the axis, however Mercury and Jupiter have very low inclinations 
  - Mercury: 0 (Very strange, I wonder why this is. Could it be due to the proximity to the sun?)
  - Jupiter: 3.08º, relatively low 
- Anormal Rotational Periods 
  - Venus spins very slowly, 243 times slower than Earth 
  - So does Mercury, 58.65 times slower than Earth
  - The Sun is also fairly slow, at 25.05 days. This makes sense, as it is much larger than Earth. 

Side Note: To more rigorously determine anomalies, it would be best to slap each column in a PDF and set a firm limit on how unlikely a measurement has to be to be considered an "anomaly". I would place this limit somewhere around 2σ. 

In [33]:
#Campus data 
campus_df = pd.read_csv('../Data/CampusDistances.csv')

local_distances = {}

#Calculate straight line distances between the reference point and the other points 
for index1, row1 in campus_df.iterrows():
  #Calculate the distance 
  distance = np.sqrt((row1['x'])**2 + (row1['y'])**2)

  #Find the inclination angles from the reference point
  angle = np.arctan(row1['y']/row1['x'])

  #Print the results
  print(f"To {row1['location']}: Distance - {distance:.3f} ft. Angle - { np.degrees(angle)} degrees") 
  # Ignore the 'lower' inclination angle, I just made the distances not 0 so the angle would be defined. 
  # I know it's hacky, but it works. 

  local_distances[row1['location']] = distance

print(local_distances)

'''
The range of angles of inclination isn't as diverse as the axial tilt of the planets, but I don't see why that would be relevant. 
It would make much more sense to compare the orbital inclination of the planets, but it wasn't provided, and it's pretty much 0 for all the bodies considered here, except Pluto. 
'''

To lower: Distance - 0.001 ft. Angle - 45.0 degrees
To chapel: Distance - 1133.801 ft. Angle - 6.278805427268508 degrees
To observatory: Distance - 2656.029 ft. Angle - 7.55045236984269 degrees
To twin: Distance - 7265.783 ft. Angle - 16.404670869632152 degrees
{'lower': 0.001414213562373095, 'chapel': 1133.8011289463423, 'observatory': 2656.028990805635, 'twin': 7265.7830961294185}


"\nThe range of angles of inclination isn't as diverse as the axial tilt of the planets, but I don't see why that would be relevant. \nIt would make much more sense to compare the orbital inclination of the planets, but it wasn't provided, and it's pretty much 0 for all the bodies considered here, except Pluto. \n"

To lower: Distance - 0.001 ft. Angle - 45.0 degrees

To chapel: Distance - 1133.801 ft. Angle - 6.278805427268508 degrees

To observatory: Distance - 2656.029 ft. Angle - 7.55045236984269 degrees

To twin: Distance - 7265.783 ft. Angle - 16.404670869632152 degrees

The range of angles of inclination isn't as diverse as the axial tilt of the planets, but I don't see why that would be relevant. 
It would make much more sense to compare the orbital inclination of the planets, but it wasn't provided, and it's pretty much 0 for all the bodies considered here, except Pluto. 

In [34]:
solar_radii_feet = 2.282 * 10**9 #Feet

#Given a campus location and a planet, calculate the size of that planet and the size of the sun if they were at that location.
def calc_scale(location, planet, scaling_factor=None):
  #Get the distance from to the location 
  local_distance = local_distances[location]

  #Get the size of the planet and the distance of the planet from the sun
  planet_size = ds_dict[planet][0]
  planet_distance = ds_dict[planet][1]

  #Calculate the scaling factor 
  if not scaling_factor:
    scaling_factor = local_distance / (planet_distance * solar_radii_feet) #Convert to feet 

  #Calculate the size of the planet and the sun at the location
  planet_size = (planet_size * solar_radii_feet) * scaling_factor
  sun_size = solar_radii_feet * scaling_factor

  return planet_size, sun_size, scaling_factor

In [35]:
# Can you scale the Sun such that when you place it at the reference point, any three planets (doesn't matter which) scale to the three distances mapped above?
# If so, what is the radius of the scaled Sun in feet?

for row, index in body_df.iterrows(): 
  for key in local_distances: 
    if (key == "lower" or index['Name'] == "Sun"):
      continue
    print(f"Planet: {index['Name']}, Location: {key}")
    print(calc_scale(key, index['Name']))

Planet: Mercury, Location: chapel
(0.047397178004595655, 13.532001202017277, 5.929886591593899e-09)
Planet: Mercury, Location: observatory
(0.11103206342682957, 31.69990448816667, 1.3891281546085306e-08)
Planet: Mercury, Location: twin
(0.30373723041717493, 86.71766421840738, 3.800072928063426e-08)
Planet: Venus, Location: chapel
(0.0637047351491316, 7.329833984426025, 3.2120219037800285e-09)
Planet: Venus, Location: observatory
(0.14923395213491256, 17.170781597756946, 7.524444170796208e-09)
Planet: Venus, Location: twin
(0.4082416007294938, 46.97206811830399, 2.0583728360343552e-08)
Planet: Earth, Location: chapel
(0.048338125671255165, 5.277480468786738, 2.3126557707216205e-09)
Planet: Earth, Location: observatory
(0.11323631619891941, 12.362962750385, 5.417599802973269e-09)
Planet: Earth, Location: twin
(0.30976714296198826, 33.819889045178876, 1.482028441944736e-08)
Planet: Mars, Location: chapel
(0.016920358372675432, 3.472026624201801, 1.5214840596852765e-09)
Planet: Mars, Locat

In [52]:
def calc_distance(scaling_factor, planet):
  distance = scaling_factor * ds_dict[planet][1] * solar_radii_feet
  size = ds_dict[planet][0] * solar_radii_feet * scaling_factor
  return distance, size

In [74]:
scaling_factor = (calc_scale("twin", "Neptune"))[2]

sun_size = solar_radii_feet * scaling_factor
print(f"Sun's Radius: {sun_size} feet \n")

print("The scaling factor is: ", scaling_factor)
print("The results below are presented in distance (feet), radius (feet) \n")

#Calculate the distances for each planet 
for row, index in body_df.iterrows():
  if (index['Name'] == "Sun"):
    continue
  print(index['Name'])
  print("   ", calc_distance(scaling_factor, index['Name']))

Sun's Radius: 1.1250794758875207 feet 

The scaling factor is:  4.930234337806839e-10
The results below are presented in distance (feet), radius (feet) 

Mercury
    (94.26664695577088, 0.003940702590242788)
Venus
    (174.0307328414232, 0.00977824193364057)
Earth
    (241.7093511686433, 0.010304961509048177)
Mars
    (367.39821377633785, 0.005482892267318121)
Jupiter
    (1256.8886260769452, 0.11536128123957977)
Saturn
    (2305.9072101488573, 0.096942253142504)
Uranus
    (4635.985355414579, 0.041295784134487666)
Neptune
    (7265.7830961294185, 0.040011299280349494)
Pluto
    (9533.016810091292, 0.0018580598518979933)


I was feeling that special kind of lazy where doing the tedious, yet simple thing 21 times seemed way easier than writing a script to do it for me. Thats why I tested all of the planets at each of the locations, and Neptune at Twin Peaks seemed to give me planets closest to the locations. 

Jupiter: Distance is 10% farther away than the chapel 
Saturn: Distance is 13% closer than the observatory 
Neptune: 0% diference in distance 

